astra/src/main/java/com/slack/astra/clusterManager/CacheNodeAssignmentService.java

```diff
@@ -194,6 +195,7 @@ protected void assignReplicasToCacheNodes() {
       // snapshots
       List<SnapshotMetadata> unassignedSnapshots =
           getUnassignedSnapshots(snapshotsWithReplicas, assignedSnapshots);
+      unassignedSnapshots.sort(Comparator.comparing(a -> a.snapshotId));
```

Comment:
Why do we sort by `a.snapshotId?` If we want to randomize this list we could instead use Collections.shuffle?

Reply:
As discussed via Slack, we're looking to ensure consistent assignments across runs at this point.